QuickFIXn/Logger/ScreenLog.cs

```diff
@@ -56,16 +60,29 @@ public void OnEvent(string s)
     }
     #endregion
 
-    #region IDisposable implementation
-    public void Dispose()
-    {
-        Dispose(true);
-        System.GC.SuppressFinalize(this);
-    }
-    protected virtual void Dispose(bool disposing)
+    public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception? exception,
+        Func<TState, Exception?, string> formatter)
     {
-        // Nothing to dispose of...
+        if (!IsEnabled(logLevel)) return;
+        if (eventId == LogEventIds.IncomingMessage && _logIncoming)
+        {
+            Console.WriteLine($"<incoming> {formatter(state, exception).Replace(Message.SOH, '|')}");
+        }
+        else if (eventId == LogEventIds.OutgoingMessage && _logOutgoing)
+        {
+            Console.WriteLine($"<outgoing> {formatter(state, exception).Replace(Message.SOH, '|')}");
+        }
+        else if (_logEvent)
+        {
+            Console.WriteLine($"<event> {formatter(state, exception)}");
+        }
     }
-    ~ScreenLog() => Dispose(false);
-    #endregion
+
+    public bool IsEnabled(LogLevel logLevel) => logLevel != LogLevel.None;
+
+#pragma warning disable CS8633
+    public IDisposable BeginScope<TState>(TState state) where TState : notnull => default!;
+#pragma warning restore CS8633
```

Comment:
any reason to disable the warning instead of fixing the generic constraint? could also implement explicitly

```suggestion
    IDisposable ILogger.BeginScope<TState>(TState state) => default!;
```

Reply:
I was so frustrated at not getting it to work at the time but it was because the type of the method had changed between v6 and 7 and I was implementing 7 but the MEL target was 6.